src/main/java/FileSaving.java

```diff
@@ -50,11 +50,20 @@ public List<Task> readFromFile(String fileName) {
             String type = working.getString("Type");
             String translatedType = TaskFactory.getTranslation(type);
 
-            Task newTask = TaskFactory.getInstance().createTask(translatedType);
-            newTask.fromJSONObject(working);
-
-            task.add(newTask); //@FIXME Are tasks saved sequentially? - Alex
+            if(translatedType != null){
+                Task newTask = TaskFactory.getInstance().createTask(translatedType);
+                if (newTask != null) {
+                    newTask.fromJSONObject(working);
+                     //@FIXME Are tasks saved sequentially? - Alex
+                    task.add(newTask);
+                } else {
+                    System.err.println("Failed to create task of type: " + translatedType);
+                }
+            } else {
+                System.err.println("No translation available for type: " + type);
+            }
         }
+        
         return task;
```

Comment:
It looks like it is still going to return the list even though it fails a read. The user should be notified about this and client.java should handle it

Reply:
it returns the list of tasks that arent a null type. or at least it should

Reply:
also it should just rerun the menu if it doesn't run